File: src/main/java/io/github/onepieceapi/controllers/ChapterController.java
package io.github.onepieceapi.controllers;

import static io.github.onepieceapi.constants.Urls.ALL_URL;
import static io.github.onepieceapi.constants.Urls.CHAPTER_URL;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import io.github.onepieceapi.constants.Urls;
import io.github.onepieceapi.model.Character;
import io.github.onepieceapi.services.CharacterService;
import lombok.AllArgsConstructor;

@RestController
@RequestMapping(CHAPTER_URL)
@AllArgsConstructor
public class ChapterController {
    private final CharacterService characterService;

    @GetMapping(ALL_URL)
    public ResponseEntity<Page<Character>> getCharacter(Pageable pageable) {
        return ResponseEntity.ok(characterService.getAll(pageable));
    }
}

File: src/main/kotlin/io/github/onePieceApi/controllers/ChapterController.kt
package io.github.onePieceApi.controllers

import io.github.onePieceApi.constants.Urls.ALL_URL
import io.github.onePieceApi.constants.Urls.CHAPTER_URL
import io.github.onePieceApi.model.Chapter
import io.github.onePieceApi.services.ChapterService
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(CHAPTER_URL)
class ChapterController constructor(
    private val chapterService: ChapterService
) {

    @GetMapping(ALL_URL)
    fun getCharacter(pageable: Pageable): ResponseEntity<Page<Chapter>> {
        val result = chapterService.getAll(pageable)
        return ResponseEntity.ok(result)
    }
}
